frontend/src/pages/connectionTypes/manage/advanced/DropdownAdvancedPropertiesForm.tsx

```diff
@@ -0,0 +1,201 @@
+import * as React from 'react';
+import { Button, Flex, FlexItem, FormGroup, Radio, TextInput } from '@patternfly/react-core';
+import { MinusCircleIcon, PlusCircleIcon } from '@patternfly/react-icons';
+import { Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import text from '@patternfly/react-styles/css/utilities/Text/text';
+import useDraggableTableControlled from '~/utilities/useDraggableTableControlled';
+import { DropdownField } from '~/concepts/connectionTypes/types';
+import { AdvancedFieldProps } from '~/pages/connectionTypes/manage/advanced/types';
+import ExpandableFormSection from '~/components/ExpandableFormSection';
+
+const DropdownAdvancedPropertiesForm: React.FC<AdvancedFieldProps<DropdownField>> = ({
+  properties,
+  onChange,
+  onValidate,
+}) => {
+  React.useEffect(() => {
+    if (!properties.variant) {
+      onChange({ ...properties, variant: 'single' });
+    }
+  });
```

Comment:
So yeah I was going to at first, but there was a lint error `React Hook React.useEffect has missing dependencies: 'onChange' and 'properties'. Either include them or remove the dependency array. If 'onChange' changes too often, find the parent component that defines it and wrap that definition in useCallback.` 

I looked into why this is linted, and when having an empty array, the component is less resilient with what props get sent in vs how it behaves. So with [] it would have different behavior based on identical props. But by remove the [] and having it always run means the behavior is always the same.

I'd have to add a lint exception or add `[onChange, properties]`